site/docs/transform/impute.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: docs
+title: Impute
+permalink: /docs/impute.html
+---
+
+To impute data in Vega-Lite, users can either use use an `impute` property of an [encoding field definition](#encoding) or an `impute` transform inside the [`transform`](#transform) array.
+
+## Documentation Overview
+{:.no_toc}
+
+- TOC
+{:toc}
+
+{:#encoding}
+## Impute in Encoding Field Definition
+
+{: .suppress-error}
+```json
+// A Single View Specification
+{
+  "data": ... ,
+  "mark": ... ,
+  "encoding": {
+    "x": {
+      "field": ...,
+      "type": "quantitative",
+      "impute": ...,               // Impute
+      ...
+    },
+    "y": ...,
+    ...
+  },
+  ...
+}
+```
+
+The `impute` property of [a field definition](encoding.html#field-def) can be used to generate new data objects in place of the missing data.
+
+{% include table.html props="value,method,frame" source="ImputeParams" %}
+
+In this example, the data object where `"x"` is `3` and `"c"` is `1` is missing. So we can impute the missing data and give the imputed field a value of `0`.
+
+<div class="vl-example" data-name="line_impute_value"></div>
+
+We can also use a `method` on existing data points to generate the missing data.
+Values are automatically grouped by the fields of some of the Encoding Channels if specified viz.: `color`, `opacity`, `fill`, `stroke`, `opacity`, `size`, `key`, `detail`.
```

Comment:
Never use the term "viz." as it's an _informal_ abbreviation of visualization.  Better just use "visualization".

Reply:
> Encoding Channels

Reply:
"viz." as in [namely](https://dictionary.cambridge.org/dictionary/english/viz)

Reply:
> the fields of some of the Encoding Channels

How are these "some" determined?  Given the current text, users will perceive this as a black box. 

I guess you group by all non-position channels or something? 
Better read the path grouping logic and summarize the important details here. 

In a separate PR, you should take this chance to help revise `line.md` as well as I suspect if the following text there is still accurate (it was written in the old days)

> Adding a field to a mark property channel such as color groups data points into different series, producing a multi-series colored line chart.

Reply:
> "viz." as in namely

Ah I see. The sentence still reads pretty confusing though. 
Instead of listing all the channels and say "some".  Give them an intuition what do these channels have in common.  Moreover, listing all channels isn't a sustainable strategy as we will have more channels in the future. Your exhaustive list here will then be outdated.